Source/Core/DiscIO/VolumeVerifier.cpp

```diff
@@ -709,70 +726,124 @@ void VolumeVerifier::Process()
   else if (m_block_index < m_blocks.size() && m_blocks[m_block_index].offset == m_progress)
   {
     bytes_to_read = VolumeWii::BLOCK_TOTAL_SIZE;
+    block_read = true;
   }
   else if (m_block_index < m_blocks.size() && m_blocks[m_block_index].offset > m_progress)
   {
     bytes_to_read = std::min(bytes_to_read, m_blocks[m_block_index].offset - m_progress);
   }
   bytes_to_read = std::min(bytes_to_read, m_max_progress - m_progress);
 
+  bool read_succeeded = false;
```

Comment:
You may want to try moving this chunk of code to a function. It'd be really easy to accidentally use `data` in it's moved from state in future changes given how basic of a name it is, and that it's in the middle of quite a large function